Add MongoDB integration

MostafaMohamed2002 opened the pull request:
- Add Kotlin coroutines and MongoDB driver dependencies  
- Create MongoDB module for Koin dependency injection  
- Configure MongoDB client with connection string  
- Register MongoDB module in application startup 
- Create the data/mongo directory where MongoDB data provider implementations   
will be placed

MostafaMohamed2002 committed:
add MongoDB integration

MostafaMohamed2002 committed:
Add directory structure for MongoDB implementations

K-M-9 commented on src/main/kotlin/di/MongoModule.kt:
```diff
@@ -0,0 +1,23 @@
+
+        val client = MongoClient.create(settings)
+        client.getDatabase(databaseName)
+    }
```
Separate the MongoDB client setup into its own class, and move all sensitive configuration such as the connection URI and database name into a properties file so they’re not accidentally pushed to GitHub.

MostafaMohamed2002 replied:
why should i seperate it to new class ?

K-M-9 reviewed (commented):
- Separate the MongoDB client setup into its own class, and move all sensitive configuration such as the connection URI and database name into a properties file so they’re not accidentally pushed to GitHub.

MostafaMohamed2002 committed:
Separate MongoDB client setup into dedicated class and secure credentials

- Created MongoDBClient class to encapsulate database connection logic
- Moved sensitive credentials (username, password) to local.properties
- Updated MongoModule to read credentials from local.properties
- Added local.properties to .gitignore to prevent accidental commits

MostafaMohamed2002 commented:
I separated MongoDB client setup into a dedicated class to improve code organization and make it easier to manage database connections. This makes the code more maintainable and reusable, while keeping database connection logic in one place instead of scattered throughout the application.

MostafaMohamed2002 committed:
Update properties file loading method

- Change from ClassLoader to direct File reading for local.properties
- Improve error messages for missing credentials
